### PR TITLE
feat: add helper function `defineLazyMountComponent`

### PR DESCRIPTION
**Pull request by Mini-ghost**
Modal are often not opened immediately when entering a website. They usually require some action, like click button to trigger their appearance.

To handle this, we can use `defineAsyncComponent` to load the modal component dynamically:

```vue
<script setup lang="ts">
import { defineAsyncComponent, ref } from 'vue'

const LazyCustomModal = defineAsyncComponent(() => import('@/components/CustomModal.vue'));

const isActive = ref(false)
</script>

<template>
  <LazyCustomModal v-model="isActive"  />
</template>
```

However, this approach only splits the component into a separate chunk, which will still be loaded when entering the page, even if we don't open it.

To achieve delayed loading, we can implement the following:

```vue
<script setup lang="ts">
import { defineAsyncComponent, ref } from 'vue'

const LazyCustomModal = defineAsyncComponent(() => import('@/components/CustomModal.vue'));

const isActive = ref(false)
</script>

<template>
  <template v-if="isActive">
    <LazyCustomModal v-model="isActive"  />
  </template>
</template>
```

But a problem with this method is that the component will be unloaded when the modal is closed. So we need to do something special to handle this:

```vue
<script setup lang="ts">
import { defineAsyncComponent, ref } from 'vue'

const LazyCustomModal = defineAsyncComponent(() => import('@/components/CustomModal.vue'));

const isActive = ref(false)
const isModalMounted = ref(false)
</script>

<template>
  <template v-if="isActive || isModalMounted">
    <LazyCustomModal 
      v-model="isActive"
      @vnode-mounted="isModalMounted = true"
    />
  </template>
</template>
```

This solution might seem a bit wordy, so I attempted to add a new method called `defineLazyMountCompone` to encapsulate the above logic:

```vue
<script setup lang="ts">
import { ref } from 'vue'
import { defineLazyMountComponent } from 'vue-final-modal'

const LazyCustomModal = defineLazyMountComponent(() => import('@/components/CustomModal.vue'));

const isActive = ref(false)
</script>

<template>
  <LazyCustomModal v-model="isActive"  />
</template>
```

I hope this simplified explanation is easier to understand. If you have any questions, feel free to ask!

### CODE REVIEW

**Comment by hunterliu1003** [comment]
@Mini-ghost
May I ask you to add at least one use case for `defineLazyMountComponent()` inside of the `/viteplay`?
I like the `defineLazyMountComponent()`, but I think it should be more easy to use the function. Is it possible to have a `lazy` prop supported by `VueFinalModal.vue`?

**Comment by Mini-ghost** [comment]
@hunterliu1003 In my case, I often encapsulate modals into components, such as drawers. For these kinds of components, I won't use `useModal`, but I will explore how to integrate them.

**Comment by Mini-ghost** [comment]
Usually `useModal` I would use with asnyc component, it will be loaded as needed, so I think `useModal` is not needed when using `defineLazyMountComponent`, `defineAsyncComponent` is enough
